sdk-installer/src/run.py

```diff
@@ -80,69 +155,101 @@ def get_tag_or_branch_name(tag_or_branch_name: str) -> str:
+        return Conan(verbose_logging)
+    elif programming_language == "python":
+        return Pip(verbose_logging)
 
-    Args:
-        verbose (bool): Enable verbose logging.
-    """
-    lang = get_programming_language()
-    if lang not in SUPPORTED_LANGUAGES:
-        print("gRPC interface not yet supported for programming language " f"{lang!r}")
-        return
+    raise RuntimeError(f"No package manager available for {programming_language!r}!")
+
+
+def force_clone_repo(
+    git_url: str, git_ref: str, output_dir: str, verbose_logging: bool
+) -> None:
+    if os.path.exists(output_dir):
+        shutil.rmtree(output_dir)
+
+    subprocess.check_call(
+        ["git", "clone", "--depth", "1", "-b", git_ref, git_url, output_dir],
+        stdout=subprocess.DEVNULL if not verbose_logging else None,
+    )
+
+    subprocess.check_call(
+        ["git", "config", "--global", "--add", "safe.directory", output_dir],
+        stdout=subprocess.DEVNULL if not verbose_logging else None,
+    )
 
+
+def install_packge_if_required(
+    package_dict: Dict[str, str], lang: str, verbose_logging: bool
+):
     required_sdk_version: Optional[str] = None
-    sdk_install_path = os.path.join(get_project_cache_dir(), f"vehicle-app-{lang}-sdk")
-    git_url = f"https://github.com/eclipse-velocitas/vehicle-app-{lang}-sdk.git"  # noqa: E231
+    package_clone_path = os.path.join(
+        get_project_cache_dir(), f"{package_dict['id']}-{lang}"
+    )
 
-    if lang == "cpp":
-        required_sdk_version = get_required_sdk_version_cpp()
-    elif lang == "python":
-        required_sdk_version = get_required_sdk_version_python()
+    package_manager = get_package_manager(lang, verbose_logging)
+    required_sdk_version = package_manager.get_required_package_version(
+        package_dict["id"]
+    )
 
     if required_sdk_version is None:
         print("No SDK dependency detected -> Skipping installation.")
         return
 
-    print(f"Installing version {required_sdk_version!r}...")
+    if package_manager.is_package_installed(package_dict["id"], required_sdk_version):
+        print("Correct version of SDK already installed!")
+        return
 
-    if os.path.exists(sdk_install_path):
-        shutil.rmtree(sdk_install_path)
+    # clone git repository which contains SDK package
+    git_url = package_dict["gitRepo"]
+    project_variables = ProjectVariables(os.environ)
+    git_url = project_variables.replace_occurrences(git_url)
+    git_ref = package_dict["gitRef"]
+    if git_ref == "auto":
+        git_ref = get_tag_or_branch_name(required_sdk_version)
 
-    subprocess.check_call(
-        [
-            "git",
-            "clone",
-            "--depth",
-            "1",
-            "-b",
-            get_tag_or_branch_name(required_sdk_version),
-            git_url,
-        ],
-        cwd=get_project_cache_dir(),
-    )
+    force_clone_repo(git_url, git_ref, package_clone_path, verbose_logging)
```

Comment:
will we also have a normal clone or something? if not we should rename this method to just clone_repo

Reply:
Force clone will remove any previously created directory at the package_clone_path, hence the force.